src/erlang_ls_protocol_impl.erl

```diff
@@ -0,0 +1,196 @@
+-module(erlang_ls_protocol_impl).
+
+-export([ initialize/1
+        , initialized/1
+        , shutdown/1
+        , exit/1
+        ]).
+
+-export([ textdocument_didopen/1
+        , textdocument_didchange/1
+        , textdocument_didsave/1
+        , textdocument_didclose/1
+        , textdocument_hover/1
+        , textdocument_completion/1
+        , textdocument_definition/1
+        ]).
+
+%%==============================================================================
+%% Macros
+%%==============================================================================
+-define(DEFAULT_CONFIG_PATH, "erlang_ls.config").
+
+%%==============================================================================
+%% Initialize
+%%==============================================================================
+
+-spec initialize(map()) ->
+  {response, map()} | {} | {notification, binary(), map()}.
+initialize(Params) ->
+  #{ <<"rootUri">> := RootUri
+   , <<"initializationOptions">> := InitOptions
+   } = Params,
+  ok = erlang_ls_buffer_server:set_root_uri(RootUri),
+  Config = consult_config(filename:join([ erlang_ls_uri:path(RootUri)
+                                        , config_path(InitOptions)
+                                        ])),
+  OtpPath = maps:get("otp_path", Config, code:root_dir()),
+  DepsDirs = maps:get("deps_dirs", Config, []),
+  ok = erlang_ls_buffer_server:set_otp_path(OtpPath),
+  ok = erlang_ls_buffer_server:set_deps_dirs(DepsDirs),
+  Result = #{ capabilities =>
+                #{ hoverProvider => false
+                 , completionProvider =>
+                     #{ resolveProvider => false
+                      , triggerCharacters => [<<":">>, <<"#">>]
+                      }
+                 , textDocumentSync => 1
+                 , definitionProvider => true
+                 }
+            },
+  {response, Result}.
+
+-spec config_path(map()) -> erlang_ls_uri:path().
+config_path(#{<<"erlang">> := #{<<"config_path">> := ConfigPath}}) ->
+  ConfigPath;
+config_path(_) ->
+  ?DEFAULT_CONFIG_PATH.
+
+-spec consult_config(erlang_ls_uri:path()) -> map().
```

Comment:
Note (outside of the scope for this PR): we could move config handling to its own separate module.